### ISSUE TITLE
Model pred seems bad when trying to fit on only one sample

### ISSUE THREAD

**Issue by Wang-Yu-Qing** [NikolasEnt/soccernet-calibration-sportlight#4]
Hi Nikolas. 

I'm learning this repo and because of the limited computation resource I have, I only test the keypoints detection model upon **one input sample**, which is 00000.jpg. I want to see if the model can fit this single sample well. So I made the dataloader only read this single sample to train the model. 

And after around 80 epochs, I saved the prediction masks with adaptive wing loss of value 0.01495 and compared it with the label masks. It seems that the result is bad:

The below code produce the raw image and the label masks:
```
import cv2
import numpy as np
import matplotlib.pyplot as plt

def show_heatmap(img, heatmaps):
  # accumulate all masks apart from empty class
  acc_heatmap = heatmaps[:-1, :, :].max(axis=0)
  # heatmap for empty class
  empty_class_heatmap = heatmaps[-1, :, :]
  
  fig, ax = plt.subplots(nrows=1, ncols=3)
  ax[0].imshow(img, cmap="gray")
  ax[1].imshow(acc_heatmap, cmap="gray")
  ax[2].imshow(empty_class_heatmap, cmap="gray")
  
  plt.show()

img = cv2.imread('../dataset/calibration-2023/train_v2/00000.jpg') # BGR
# this is the label masks I saved during training
hm = np.load('model_ckpt/heatmap.npy')[0]
show_heatmap(img, hm)
```
![image](https://github.com/NikolasEnt/soccernet-calibration-sportlight/assets/19722776/3ad3842b-73d7-46c8-adbf-d5268ee51bc8)

And the below code is to produce the pred masks:
```
img = cv2.imread('../dataset/calibration-2023/train_v2/00000.jpg') # BGR
# this is the masks preded by the trained model saved during training 
pred = np.load('model_ckpt/pred.npy')[0]
show_heatmap(img, pred)
```
![image](https://github.com/NikolasEnt/soccernet-calibration-sportlight/assets/19722776/4c2dc35c-3fea-4564-a79f-567f81e461bc)

It seems that the result is not good even with this one single sample. Is a awing loss of 0.01495 poor or am I missing some important things? Thanks!

**Comment by Wang-Yu-Qing** [comment]
@NikolasEnt You are right. I can produce a good result now.
